cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -103,6 +106,28 @@ func PerformPostUpgradeTasks(client clientset.Interface, cfg *kubeadmapi.InitCon
 		errs = append(errs, err)
 	}
 
+	if err := PerformAddonsUpgrade(client, cfg, out); err != nil {
+		errs = append(errs, err)
+	}
+
+	return errorsutil.NewAggregate(errs)
+}
+
+// PerformAddonsUpgrade performs the upgrade of the coredns and kube-proxy addons
+func PerformAddonsUpgrade(client clientset.Interface, cfg *kubeadmapi.InitConfiguration, out io.Writer) error {
+	if features.Enabled(cfg.FeatureGates, features.UpgradeAddonsAfterControlPlane) {
+		allUpgraded, unupgradedNodes, err := allControlPlaneInstancesUpgraded(client, cfg.NodeRegistration.Name)
```

Comment:
`err == nil` and `Len(unupgradedNodes)>0` will imply the  `!allUpgraded`,  you can reduce one return value, just a soft suggestion.

Reply:
Good catch, updated.